prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -2417,13 +2489,50 @@ int rxr_pkt_proc_dc_write_rta(struct rxr_ep *ep,
 	return ret;
 }
 
+static int rxr_cuda_fetch_atomic(struct efa_mr *efa_mr, struct iovec *iov, char* data, void* result, size_t dtsize, int op, int dt)
+{
+	char *host_data;
+	uint64_t device = efa_mr->peer.device.reserved;
+	int ret;
+
+	/* Step 1: Copy data from device to temporary host buffer */
+	host_data = malloc(iov->iov_len);
+	if (!host_data) {
+		return FI_ENOSPC;
+	}
```

Comment:
future optimization: use some sort of memory pool so we don't continually alloc/free host memory for this.

Reply:
I'm guessing `iov->iov_len` is dynamic (and we want to avoid VLAs). Otherwise it makes more sense to just have the string buffer be stack-allocated. E.g.
```c
char host_data[iov->iov_len];
```

Reply:
good idea.. switched to VLA's.